test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+    "CardPaymentProcessor: payment with the provided authorization ID has an inappropriate status";
+  const REVERT_MESSAGE_IF_PAYMENT_REVOCATION_COUNTER_HAS_REACHED_MAXIMUM =
+    "CardPaymentProcessor: revocation counter of the payment has reached the configured maximum";
+
+  let cardPaymentProcessor: Contract;
+  let brlcMock: Contract;
+  let deployer: SignerWithAddress;
+  let user1: SignerWithAddress;
+  let user2: SignerWithAddress;
+
+  async function setUpContractsForPayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(brlcMock.mint(payment.account.address, payment.amount));
+      const allowance: BigNumber = await brlcMock.allowance(payment.account.address, cardPaymentProcessor.address);
+      if (allowance.lt(BigNumber.from(ethers.constants.MaxUint256))) {
+        await proveTx(brlcMock.connect(payment.account).approve(
+          cardPaymentProcessor.address,
+          ethers.constants.MaxUint256
+        ));
+      }
+    }
+  }
+
+  async function makePayments(payments: TestPayment[]) {
+    for (let payment of payments) {
+      await proveTx(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        createBytesString(payment.authorizationId, BYTES16_LENGTH),
+        createBytesString(payment.makingPaymentCorrelationId, BYTES16_LENGTH)
+      ));
+      payment.status = PaymentStatus.Uncleared;
+    }
+  }
+
+  async function addAccountToCardPaymentProcessorWhitelist(account: SignerWithAddress) {
+    await proveTx(cardPaymentProcessor.setWhitelistEnabled(true));
+    await proveTx(cardPaymentProcessor.setWhitelistAdmin(deployer.address));
+    await proveTx(cardPaymentProcessor.updateWhitelister(deployer.address, true));
+    await proveTx(cardPaymentProcessor.whitelist(account.address));
+  }
+
+  async function clearPayments(payments: TestPayment[], adminAccount: SignerWithAddress) {
+    const authorizationIds: string[] = [];
+    payments.forEach(payment => {
+      authorizationIds.push(createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      payment.status = PaymentStatus.Cleared;
+    });
+    await proveTx(cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds));
+  }
+
+  function defineBalancesPerAccount(payments: TestPayment[], targetPaymentStatus: PaymentStatus) {
+    const balancesPerAccount: Map<string, number> = new Map<string, number>();
+
+    payments.forEach(payment => {
+      const address: string = payment.account.address;
+      let newBalance: number = balancesPerAccount.get(address) || 0;
+      if (payment.status == targetPaymentStatus) {
+        newBalance += payment.amount;
+      }
+      balancesPerAccount.set(address, newBalance);
+    });
+
+    return balancesPerAccount;
+  }
+
+  function defineExpectedCardPaymentProcessorState(payments: TestPayment[]): CardPaymentProcessorState {
+    const tokenBalance: number = payments.map(function (payment: TestPayment): number {
```

Comment:
Codacy has a fix for the issue: [Insert `⏎······`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
    const tokenBalance: number = payments
```